notifications.bs

```diff
@@ -1087,8 +1143,10 @@ the same underlying <a>notification</a> of {{Notification}} objects already in e
 given <var>notification</var> and <var>action</var>,
 <a>fire an event</a> named <var>e</var>, using {{NotificationEvent}}, with the
 {{NotificationEvent/notification}} attribute initialized to a new
-{{Notification}} object representing <var>notification</var> and the
-{{NotificationEvent/action}} attribute initialized to <var>action</var>.
+{{Notification}} object representing <var>notification</var>, the
+{{NotificationEvent/action}} attribute initialized to <var>action</var>, and the
+{{NotificationEvent/reply}} attribute initialized to <var>reply</var>.
```

Comment:
Where does the reply variable come from?

Reply:
Ah good point, guess I can just change the beginning of this sentence to 'given notification, action and reply' which should cover it, yes?

Reply:
Well, you'll need to update the callers of this algorithm too, then.

Reply:
Ah yes, that makes sense.

Have updated Step 1.5 of Section 2.7 'Activating a Notification' to 'given notification, action and reply' also. I think this now covers it.

Reply:
There's two callers of this algorithm, so I don't think that quite covers it. Also, if both are going to set it to null, does it need to be a variable for this algorithm?

Reply:
Step 1.4 of Section 2.7 ('Activating a notification') sets 'reply' to the text entered by the user during activation, so it shouldn't be null in that case.

I assume you are talking about Section 2.8 ('Closing a notification') as the other caller of this algorithm. That caller does not pass anything for 'action' either right now, so I guess this needs fixing too?

Reply:
Ah yeah :/ I guess action needs to be the empty string there; @beverloo can confirm.

Reply:
Hm yeah that would involve adding a whole new interface to the ServiceWorker API, no?

I'm up for doing this but won't this involve quite a lot of work updating idls and tests etc? Could it be done as a followup in a separate change?

Reply:
Agreed with beverloo@ offline that changing the event hierarchy is out of scope for this change.

Have now updated the 'Closing a notification' algorithm as suggested to pass the empty string & null.